src/main/java/seedu/address/model/person/ReminderDatePredicate.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.model.person;
+
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Orders}'s {@code DeliveryDate} is within 3 days of the current date.
+ */
+public class ReminderDatePredicate implements Predicate<Person> {
+
+    private long days;
+
+    public ReminderDatePredicate(long days) {
+        this.days = days;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return isWithinXDays(person);
+    }
+
+    /**
+     * Returns true if the order's delivery date is within X days of the current date.
+     */
+    public boolean isWithinXDays(Person person) {
+        LocalDate toTest = person.getDeliveryDate().getValue();
+        LocalDate dateToday = LocalDate.now();
+        LocalDate acceptableDate = dateToday.plusDays(days + 1);
+        return toTest.isEqual(dateToday) || toTest.isBefore(acceptableDate);
```

Comment:
Isn't toTest.isBefore(acceptableDate) sufficient haha? Because dateToday is before the acceptable date? And Also because remind is for upcoming deadlines right, you could consider making sure all the dates are after today. In case a person/order with a past delivery date is still in the system.

Reply:
yep, changed the conditions to be more strict.

Reply:
> Looks great!!
> Also, wanted to ask whether inputting an add/find command after a remind command changes the address book back to the original person list? Not sure if the addressbook already takes care of that somewhere else.

@pavz02 , RemindCommand acts on a different list than add command, but on the same "filtered" order list as FindCommand, it will not affect the state of the original person list.